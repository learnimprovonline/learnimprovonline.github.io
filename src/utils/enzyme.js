import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import toJson from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter() })
