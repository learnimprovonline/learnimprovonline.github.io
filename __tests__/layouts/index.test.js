import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import TemplateWrapper from '../../src/layouts'
import Helmet from 'react-helmet'
import Header from '../../src/components/header'

const site = {
  siteMetadata: {
    title: 'Learn Improv Online',
    description: 'Free improv warm-ups and exercises with high-quality written descriptions, animated instructionals, and video examples. Start learning now!',
    keywords: 'improv, learn improv, learn improv online'
  }
}

const mockChildren = jest.fn()
mockChildren.mockReturnValue(<div>Children</div>)

const layout = shallow(
  <TemplateWrapper>
    {mockChildren}
  </TemplateWrapper>
);
const helmet = layout.find(Helmet);

test('Layout contains a Helmet with the site title', () => {
  const helmetTitleText = helmet.prop('title');

  expect(helmetTitleText).toEqual(site.siteMetadata.title);
});

test('Layout contains a Helmet with the correct meta data', () => {
  const helmetMeta = helmet.prop('meta');
  const helmetDescription = helmetMeta.find(x => x.name === 'description');
  const helmetDescriptionText = helmetDescription.content;
  const helmetKeywords = helmetMeta.find(x => x.name === 'keywords');
  const helmetKeywordsText = helmetKeywords.content;

  expect(helmetDescriptionText).toEqual(site.siteMetadata.description);
  expect(helmetKeywordsText).toEqual(site.siteMetadata.keywords);
});

test('Layout contains a Header with the site title', () => {
  const header = layout.find(Header);
  const headerText = header.children().text();

  expect(headerText).toEqual(site.siteMetadata.title);
});

test('Layout renders children', () => {
  expect(mockChildren.mock.calls.length).toEqual(1);
})

test('Layout renders correctly', () => {
  let tree = toJson(layout);
  expect(tree).toMatchSnapshot();
});