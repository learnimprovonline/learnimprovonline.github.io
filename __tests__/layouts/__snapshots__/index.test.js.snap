// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Layout renders itself 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    meta={
      Array [
        Object {
          "content": "Site Description",
          "name": "description",
        },
        Object {
          "content": "one, two, three",
          "name": "keywords",
        },
      ]
    }
    title="Site Title"
  />
  <Header>
    Site Title
  </Header>
  <div>
    Children
  </div>
  <Footer />
</div>
`;
